docs/dcp2_system_design.rst

```diff
@@ -960,221 +960,91 @@ Project-level matrices
-the Data Browser to list them on its *Files* tab and make them
-discoverable and downloadable like any other data file.
+- EGA Accession Field
 
-While this is no different to how other supplementary files are handled,
-Azul provides special treatment of supplementary files that describe
-contributor-generated matrices (CGMs): each hit in Azul's
-``/index/project`` response contains a listing of all CGMs so that those
-matrices can be exposed on the Data Browser's Projects tab, both the
-list view and the project details view.
+DUO (Data Use Ontology) codes will not be included at this time.
 
-The set of properties the metadata schema currently allows for in
-``supplementary_file`` entities is limited, so the challenge is
-how to encode matrix stratification. The schema defines the following
-properties:
 
--  ``supplementary_file.provenance.submitter_id``
 
--  ``supplementary_file.file_core.format``
+``Schema changes for CGMs (dcp3)``
 
--  ``supplementary_file.file_core.content_description``
+For release 3 and onwards, we intend to store matrices provided by data contributors (as opposed to the matrices generated by the DCP) in the informationally rich metadata schema. The CGMs will be described as ‘analysis_file’ entities, similar to DCP-generated matrices, and will be linked via a process to:
+the ‘sequence_file’ OR ‘cell suspension’ entities that contributed to it
+the ‘analysis_protocol’ that generated the file
+the ‘library_preparation_protocol’ applied to the cell suspensions that generated it.
+The CGM files will be exported from Ingest as analysis files, and imported and indexed as analysis files.
 
--  ``supplementary_file.file_core.file_name``
+Four optional fields are added to the schema to facilitate storing CGMs metadata.
 
--  ``supplementary_file.file_description``
+- “analysis_file.file_core.file_source”
+    - optional field set to one of two values, contributor-generated or DCP-generated.
 
-The ``submitter_id`` field is set to one out of a few statically
-defined v5 UUIDs that designate the matrix file as having been generated
-by a contributor. The namespace for the v5 UUID is
-``382415e5-67a6-49be-8f3c-aaaa707d82db`` and the name is one of
-the values from the tracking spreadsheet column named
-``file_source`` e.g. ``contributor``, ``hca release``,
-``arrayexpress`` and so on. A complete list can
-be found in the `Azul source`_ [9]_.
+- “analysis_file.matrix.cell_count”
+    - Optional field storing the number of cells which are analyzed in the analysis file
 
-.. _Azul source: https://github.com/DataBiosphere/azul/blob/a820e259ba9e37a94b5788a257d4c6f43fe31801/src/azul/plugins/metadata/hca/transform.py#L271
+- “analysis_protocol.matrix.data_normalization”
+    - Optional descriptive field describing the processes by which the analysis data is normalized
+    - Ideally ontologised, but as there is no appropriate ontology yet we are including it as a first step, and will be re-assessed
```

Comment:
This is unfortunate, given that you touted the use of ontologies as a major boon of this approach. How difficult is it to add an ontology?

Reply:
Perhaps I didn't express what I meant about ontologies clearly previously. When I complained about not being able to use ontologies properly in the current system, I was meaning that this proposed approach will allow us to curate the existing ontology,`content_description`, accurately for each and every file. That is, curate files with their true content description rather than applying a blanket term. Bottom line being, I don't like calling things that aren't matrices, matrices. 

I think the main overall benefits of the proposed approach are that users: 
1. Have more accurate information about the matrices and other relevant analysis files 
2. Have more useful information about the matrices and other relevant analysis files
3. Understand how the matrices and analysis files are linked to other entities in the experimental graph
I don't think that having these fields as ontologies vs enums at this stage would impact on the benefits above. 

> how difficult is it to add an ontology? 
Wranglers don't have direct control over ontologies, it would be a process of collaborating with ontologists to request terms, adding them to an ontology, importing them into the HCA ontology and releasing before we can use them. I think because there isn't a clear existing ontology with the terms we would need, we went with the simpler 'enum' route as we wanted to get this moving quickly.

Reply:
I agree on benefits 1 -3. But confused about 

> Bottom line being, I don't like calling things that aren't matrices, matrices.

Do you have an example of where a file that isn't a matrix is described as a matrix?

Also confused why curation of an ontology is embraced here:

> I was meaning that this proposed approach will allow us to curate the existing ontology,content_description

but described as being more involved in the last paragraph.